question-servers/freeformPythonLib/prairielearn.py

```diff
@@ -789,11 +795,17 @@ def string_to_2darray(s, allow_complex=True):
         # Iterate over rows
         for i in range(0, m):
 
-            # Split on space
-            s_row = s[i].split()
+            # Split row
+            s_row = re.split(matlab_delimiter_regex, s[i])
+
+            # Ignore first/last token if empty string (occurs when row leads/trails with valid delimiter)
+            if s_row and not s_row[0]:
+                s_row.pop(0)
+            if s_row and not s_row[-1]:
+                s_row.pop(-1)
```

Comment:
For `re.split()` side effect:

Valid delimiter at beginning/end of input will add exactly 1 empty string to beginning/end of output (2+ if invalid)